src/content_cao.cpp

```diff
@@ -716,12 +716,26 @@ void GenericCAO::updateLightNoCheck(u8 light_at_pos)
 	}
 }
 
+m4x4f GenericCAO::getTransform()
+{
+	scene::ISceneNode *node = getSceneNode();
+	if (node)
+		return node->getAbsoluteTransformation();
+
+	return m4x4f(m4x4f::EM4CONST_IDENTITY);
+}
+
 v3s16 GenericCAO::getLightPosition()
 {
-	if (m_is_player)
-		return floatToInt(m_position + v3f(0, 0.5 * BS, 0), BS);
+	// Players and entities now both use a property to define the light sampling offset.
```

Comment:
Applying the anchor to both players and non-players will cause a change of behaviour and bugs when the default is used by mods that don't yet define the anchor. See thread for details.

Reply:
Resolved by update.